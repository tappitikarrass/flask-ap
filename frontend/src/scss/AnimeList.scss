@use "base16" as *;

#anime-list.content {
    display: flex;
    flex-direction: column;
    gap: 1em;
}

// desktop
@media screen and (min-width: 810px) {
    #anime-list.content {
        width: 70vw;
        margin-left: 15vw;
    }
}

// tablet
@media screen and (min-width: 500px) and (max-width: 810px) {
    #anime-list.content {
        width: 100vw;
        margin-left: 0vw;
    }
}

// phone
@media screen and (max-width: 500px) {
    #anime-list.content {
        width: 100vw;
        margin-left: 0;
    }
}
