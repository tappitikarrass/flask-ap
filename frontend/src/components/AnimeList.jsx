import React, { useState, useRef, useEffect } from 'react';
import { useCookies } from 'react-cookie';
import { useForm } from 'react-hook-form';
import { useParams, useNavigate } from 'react-router-dom';
import '../scss/AnimeList.scss';

export default function AnimeList() {
  const navigate = useNavigate();
  const [cookies] = useCookies(['token', 'username', 'user_id']);
  const { listId } = useParams();
  const [animeList, setAnimeList] = useState([]);

  const {
    register,
    handleSubmit,
  } = useForm();
  const addForm = useRef();

  useEffect(() => {
    async function effect() {
      const response = await fetch(
        `http://localhost/backend/list/${cookies.username}/${listId}`,
        {
          method: 'GET',
          headers: {
            Authorization: `Bearer ${cookies.token}`,
          },
        },
      );

      const json = await response.json();
      const jsonArray = [];
      jsonArray.push(json);
      setAnimeList(json);
    }
    effect();
  }, []);

  async function addAnime() {
    // fetch anime mal_id
    const queryData = new FormData(addForm.current);
    const queryDataJson = JSON.stringify(Object.fromEntries(queryData));

    const response = await fetch(
      'http://localhost/backend/anime/search/',
      {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          Authorization: `Bearer ${cookies.token}`,
        },
        body: queryDataJson,
      },
    );
    const animeData = await response.json();
    // actually add anime
    const postResponse = await fetch(
      `http://localhost/backend/list/${cookies.username}/${listId}`,
      {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          Authorization: `Bearer ${cookies.token}`,
        },
        body: JSON.stringify(
          {
            mal_id: Number(animeData.mal_id),
            list_id: Number(listId),
          },
        ),
      },
    );
    const json = await postResponse.json();
    window.location.reload();
    return json;
  }

  async function removeAnime(malId) {
    const response = await fetch(
      `http://localhost/backend/list/${cookies.username}/${listId}/${malId}`,
      {
        method: 'DELETE',
        headers: {
          Authorization: `Bearer ${cookies.token}`,
        },
      },
    );
    if (response.status === 200) {
      window.location.reload();
    }
  }

  return (
    <div
      id="anime-list"
      className="content"
    >
      <div
        id="anime-search"
        className="ui-row"
      >
        <form
          id="anime-search-form"
          ref={addForm}
          method="post"
          onSubmit={handleSubmit(addAnime)}
          acceptCharset="utf-8"
        >
          <input
            id="anime-search-bar"
            {...register('query')}
            className="field"
            type="search"
            placeholder="Enter anime name"
          />
          <button
            id="add-bt"
            type="submit"
            className="bt-blue-end"
          >
            Add
          </button>
        </form>
      </div>
      <table id="anime-table">
        <tbody>
          <tr>
            <td>MyAnimeList Id</td>
            <td />
            <td />
          </tr>
          {
            animeList
              .filter((anime) => anime.list_id === Number(listId))
              .map((value, key) => (
                <tr key={key}>
                  <td>{value.mal_id}</td>
                  <td>
                    <button
                      id="open-bt"
                      className="bt-blue-end"
                      type="submit"
                      onClick={() => {
                        navigate(`/anime/${value.mal_id}`);
                      }}
                    >
                      Open
                    </button>
                  </td>
                  <td>
                    <button
                      id="open-bt"
                      className="bt-green-end"
                      type="submit"
                      onClick={() => {
                        window.open(`https://myanimelist.net/anime/${value.mal_id}`, '_blank');
                        // navigate(`/anime/${value.mal_id}`);
                      }}
                    >
                      MyAnimeList
                    </button>
                  </td>
                  <td>
                    <button
                      id="delete-bt"
                      className="bt-red-end"
                      type="submit"
                      onClick={() => {
                        removeAnime(value.mal_id);
                      }}
                    >
                      Remove
                    </button>
                  </td>
                </tr>
              ))
          }
        </tbody>
      </table>
    </div>
  );
}
