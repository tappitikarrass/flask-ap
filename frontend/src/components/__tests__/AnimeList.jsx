import React from "react";
import { HashRouter } from "react-router-dom";
import { create, act } from "react-test-renderer";
import AnimeList from '../AnimeList';

describe('AnimeList', () => {
  it('',() => {
    let component;
    act(() => {
      component = create(
        <HashRouter>
          <AnimeList />
        </HashRouter>
      );
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
