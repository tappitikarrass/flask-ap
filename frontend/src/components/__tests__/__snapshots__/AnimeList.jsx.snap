// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimeList  1`] = `
<div
  className="content"
  id="anime-list"
>
  <div
    className="ui-row"
    id="anime-search"
  >
    <form
      acceptCharset="utf-8"
      id="anime-search-form"
      method="post"
      onSubmit={[Function]}
    >
      <input
        className="field"
        id="anime-search-bar"
        name="query"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter anime name"
        type="search"
      />
      <button
        className="bt-blue-end"
        id="add-bt"
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
  <table
    id="anime-table"
  >
    <tbody>
      <tr>
        <td>
          MyAnimeList Id
        </td>
        <td />
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;
